x-pack/plugins/apm/public/components/app/service_profiling/index.tsx

```diff
@@ -10,24 +10,24 @@ import {
   getValueTypeConfig,
   ProfilingValueType,
 } from '../../../../common/profiling';
+import { useApmServiceContext } from '../../../context/apm_service/use_apm_service_context';
 import { useUrlParams } from '../../../context/url_params_context/use_url_params';
+import { useApmParams } from '../../../hooks/use_apm_params';
 import { useFetcher } from '../../../hooks/use_fetcher';
 import { APIReturnType } from '../../../services/rest/createCallApmApi';
 import { ServiceProfilingFlamegraph } from './service_profiling_flamegraph';
 import { ServiceProfilingTimeline } from './service_profiling_timeline';
 
-interface ServiceProfilingProps {
-  serviceName: string;
-  environment?: string;
-}
-
 type ApiResponse = APIReturnType<'GET /api/apm/services/{serviceName}/profiling/timeline'>;
 const DEFAULT_DATA: ApiResponse = { profilingTimeline: [] };
 
-export function ServiceProfiling({
-  serviceName,
-  environment,
-}: ServiceProfilingProps) {
+export function ServiceProfiling() {
+  const { serviceName } = useApmServiceContext();
+
+  const {
+    query: { environment },
+  } = useApmParams('/services/:serviceName/profiling');
+
   const {
     urlParams: { kuery, start, end },
```

Comment:
Should `kuery` be coming from `useApmParams` here?

Reply:
I've tried to minimise the amount of changes, and will create a follow-up issue (or more) to get rid of `useUrlParams` entirely.